main.go

```diff
@@ -335,6 +336,32 @@ func GetTransactions(oConn *ouroboros.Connection) string {
 			}
 		}
 
+		// Check if Tx has certificates and compare against known types
+		if tx.Certificates() != nil {
+			for _, certificate := range tx.Certificates() {
+				certCbor := certificate.Cbor()
```

Comment:
Instead of decoding, you can do a type switch 

```
switch cert := certificate.(type) {
  case ledger.StakeDelegationCertificate:
    ...
  ...
}
```

There should be an example in the recent node support for stake and pool certificates

Reply:
That doesn't work without a change in gOuroboros.

```
./main.go:344:10: impossible type switch case: lcommon.StakeRegistrationCertificate
	certificate (variable of type "github.com/blinklabs-io/gouroboros/ledger/common".Certificate) cannot have dynamic type "github.com/blinklabs-io/gouroboros/ledger/common".StakeRegistrationCertificate (method Utxorpc has pointer receiver)
```